### PR TITLE
SDI-273: ✨ Allow for dual running with both ingresses

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: charts/generic-service/templates/ingress.yaml [hunk 1]
```diff
@@ -15,18 +14,14 @@ metadata:
   {{- if .Values.allowlist }}
     nginx.ingress.kubernetes.io/whitelist-source-range: {{ include "app.joinListWithComma" .Values.allowlist | quote }}
   {{- end }}
-    external-dns.alpha.kubernetes.io/set-identifier: {{ $ingressName }}-{{ .Release.Namespace }}-green
-  {{- if not .Values.ingress.v1_2_enabled }}
+    external-dns.alpha.kubernetes.io/set-identifier: {{ $fullName }}-{{ .Release.Namespace }}-green
+    external-dns.alpha.kubernetes.io/aws-weight: "0"
```
Comment (petergphillips) [inline charts/generic-service/templates/ingress.yaml:18]: weighting moved back here so can override to 0 in this file and 100 in the new ingress

File: charts/generic-service/templates/ingress_v1_2.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    {{- include "generic-service.labels" . | nindent 4 }}
+  annotations:
+  {{- with .Values.ingress.annotations }}
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  {{- if .Values.allowlist }}
+    nginx.ingress.kubernetes.io/whitelist-source-range: {{ include "app.joinListWithComma" .Values.allowlist | quote }}
+  {{- end }}
+    external-dns.alpha.kubernetes.io/set-identifier: {{ $ingressName }}-{{ .Release.Namespace }}-green
+    external-dns.alpha.kubernetes.io/aws-weight: "100"
```
Comment (petergphillips) [inline charts/generic-service/templates/ingress_v1_2.yaml:19]: weight set to 100

File: charts/generic-service/templates/ingress_v1_2.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+{{- if and .Values.ingress.enabled .Values.ingress.v1_2_enabled -}}
+{{- $fullName := include "generic-service.fullname" . -}}
+{{- $ingressName := print $fullName "-v1-2" }}
```
Comment (petergphillips) [inline charts/generic-service/templates/ingress_v1_2.yaml:3]: new name for ingress

File: charts/generic-service/templates/ingress_v1_2.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    nginx.ingress.kubernetes.io/whitelist-source-range: {{ include "app.joinListWithComma" .Values.allowlist | quote }}
+  {{- end }}
+    external-dns.alpha.kubernetes.io/set-identifier: {{ $ingressName }}-{{ .Release.Namespace }}-green
+    external-dns.alpha.kubernetes.io/aws-weight: "100"
+  {{- if .Values.ingress.modsecurity_enabled }}
+    nginx.ingress.kubernetes.io/enable-modsecurity: "true"
+    nginx.ingress.kubernetes.io/modsecurity-snippet: {{ tpl (toYaml .Values.ingress.modsecurity_snippet) . | nindent 6 }}
+  {{- end }}
+spec:
+  ingressClassName: {{- if .Values.ingress.modsecurity_enabled }} "modsec" {{- else }} "default" {{- end }}
```
Comment (petergphillips) [inline charts/generic-service/templates/ingress_v1_2.yaml:25]: using new ingress class annotation

File: charts/generic-service/values.yaml [hunk 5]
```diff
@@ -47,9 +47,9 @@ ingress:
   # instructions on advised upgrade process - https://mojdt.slack.com/archives/CH6D099DF/p1657794137979789
   # switching this to true with perform a big bang ingress switch with possible downtime as dns changes
   v1_2_enabled: false
+  v0_47_enabled: true
```
Comment (petergphillips) [inline charts/generic-service/values.yaml:50]: default is existing behaviour - so existing ingress only defined, although with weighting of `0` which has no impact if only one ingress.

File: charts/generic-service/templates/ingress.yaml (hunk 1 above)
Comment (mikehalmamoj) [reply charts/generic-service/templates/ingress.yaml:18]: If people keep the default then they won't get the new ingress - what happens with this weight? Is it ignored?

Comment (ushkarev) [reply charts/generic-service/templates/ingress.yaml:18]: the CP user guide says that if the _only_ ingress has a weight of 0 it still gets all the traffic

**Review by mikehalmamoj — APPROVED** [review]

File: charts/generic-service/templates/ingress.yaml (hunk 1 above)
Comment (petergphillips) [reply charts/generic-service/templates/ingress.yaml:18]: indeed.  From https://user-guide.cloud-platform.service.justice.gov.uk/documentation/other-topics/Switch-ingress-to-v1-ingress-controller.html#step-1-changes-to-your-current-ingress-resource:
> You must set external-dns.alpha.kubernetes.io/aws-weight value to “0” for the current ingress. This does not mean traffic will stop, 0 is a valid value for a single ingress.

**Pull request merged**